xdsl/irdl.py

```diff
@@ -1639,14 +1644,22 @@ def from_pyrdl(pyrdl_def: type[ParametrizedAttribute]) -> ParamAttrDef:
         attrdict = {**attrdict, **Generic.__dict__, **GenericData.__dict__}
 
         # Check that all fields of the attribute definition are either already
-        # in ParametrizedAttribute, or are class functions or methods.
+        # in ParametrizedAttribute, or are class functions, methods or variables.
+        type_hints: Any | None = None
         for field_name, value in clsdict.items():
             if field_name in attrdict or field_name == "name":
                 continue
             if isinstance(
                 value, (FunctionType, PropertyType, classmethod, staticmethod)
             ):
                 continue
+            # fetch type hints if not already present
+            if type_hints is None:
```

Comment:
Can we compute this outside of the loop at all time, then pass it to irdl_param_attr_get_param_type_hints?

Reply:
I'm not sure, I did it this way to have as few calls to `get_type_hints` as possible. It's only calculated once per class that has a ClassVar that is not `name`.

I don't know enough about pyrdl infrastructure though to know if it's already called somewhere else. If you have a hint I'm happy to incorporate that

Reply:
`irdl_param_attr_get_param_type_hints` is the only place where this should be called for attributes. So if you pass it already there, we will compute it once rather than twice in the worse case.
For operations, it should already be accessible in the `from_pyrdl` function.